src/Pimple/PimpleContainerAdapter.php

```diff
@@ -68,17 +68,22 @@ private function addServiceToContainer(ServiceDefinition $definition)
         $this->container[$definition->getName()] = $factory;
     }
 
+    /**
+     * @param array $arguments
+     *
+     * @return array
+     */
     private function resolveArguments(array $arguments)
     {
-        return array_map(
-            function ($argument) {
-                if (isset($this->container[$argument])) {
-                    return $this->container[$argument];
-                }
-
-                return $argument;
-            },
-            $arguments
-        );
+        return array_map([$this, 'resolveArgument'], $arguments);
```

Comment:
@localheinz 

I use this library extensively in projects and I'm a big user of map/reduce/filter. That said, I'm not sure it's helpful here as it's only used twice. Let's leave it for now.